src/Castle.Core.Tests/ComInterfacesTests.cs

```diff
@@ -15,14 +15,24 @@
 #if FEATURE_TEST_COM && !__MonoCS__ // Avoid loading adodb.dll
 namespace Castle.DynamicProxy.Tests
 {
+	using System;
 	using System.Reflection;
+	using System.Runtime.InteropServices;
+
 	using ADODB;
+
 	using NUnit.Framework;
 
 	[TestFixture]
 	public class ComInterfacesTests:BasePEVerifyTestCase
 	{
 		[Test]
+		public void Marshal_Release_throws_when_called_with_IntPtr_Zero()
+		{
+			Assert.Throws<ArgumentNullException>(() => Marshal.Release(IntPtr.Zero));
+		}
```

Comment:
This test doesn't really add anything other than documentation, however it is useful because this type of call in C++ would just return 0 references left as exceptions really are for exceptional situations (e.g. OOM, AV). I had actually thought Marshal.Release wouldn't throw with a null pointer, but it does.